dingo/core/utils/pt_to_hdf5.py

```diff
@@ -0,0 +1,77 @@
+import os
+import torch
+import numpy as np
+import h5py
+import json
+import argparse
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Convert the weights of a trained PyTorch model to HDF5 format"
+        " for distribution in the LVK's CVMFS.",
+        epilog="Training history (optimizer_state_dict) is discarded.")
+    parser.add_argument("-i", "--in_file", type=str, required=True,
+            help='Input model ".pt" weights file')
+    parser.add_argument("-o", "--out_file", type=str, required=True,
+            help='Output model ".hdf5" weights file')
+    parser.add_argument("-n", "--version_number", type=int, required=True,
+            help="Model version number (integer). "
+            "Will be included in the output filename and metadata.")
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    if os.path.splitext(args.in_file)[-1] != '.pt':
+        raise ValueError('Expected a .pt input file')
+    if os.path.splitext(args.out_file)[-1] != '.hdf5':
+        raise ValueError('Expected a .hdf5 output file')
+
+    # Build output filename with version number
+    # This is required for use on CVMFS
+    version_number = args.version_number
+    root, ext = os.path.splitext(args.out_file)
+    out_file_name = f'{root}_v{version_number}{ext}'
+    print('Output will be written to', out_file_name)
+
+    # Load data into CPU memory since we'll be saving it using CPU libraries
+    d = torch.load(args.in_file, map_location=torch.device("cpu"))
+
+    # Collect the names of the dicts that can be serialized to JSON; model_state_dict and
+    # optimizer_state_dict contain torch.tensors and cannot be JSON serialized
+    dicts_to_serialize = ['model_kwargs', 'epoch', 'metadata', 'optimizer_kwargs',
+            'scheduler_kwargs', 'scheduler_state_dict']
```

Comment:
Is this everything? What about "version" (the Dingo version, which is now saved with the model)? Do we need the "scheduler_state_dict" if we are only doing inference?

Reply:
We have to keep in mind to update this if we change the dicts saved in the `PosteriorModel` class. (One could also use a `DingoDataset` for saving, which tracks which keys to save, but that is a pretty big change from the present implementation, so I would not suggest it.)

Reply:
You're right, I think we can drop `'optimizer_kwargs', 'scheduler_kwargs', 'scheduler_state_dict'`.
Yes, any new dicts we add in `PosteriorModel` would need to be added to this list.